euphonic/spectra.py

```diff
@@ -1923,17 +1951,49 @@ def z_data(self, value: Quantity) -> None:
         self.z_data_unit = str(value.units)
         self._z_data = value.to(self._internal_z_data_unit).magnitude
 
+    @staticmethod
+    def _from_spectra_data_check(
+            spectrum_0_data_units, spectrum_i_data_units,
+            x_tick_labels, spectrum_i_x_tick_labels
+    ) -> None:
+        """Check spectrum data units and x_tick_labels are consistent"""
+        if spectrum_0_data_units != spectrum_i_data_units:
+            raise ValueError("Spectrum units in sequence are inconsistent")
+        if x_tick_labels != spectrum_i_x_tick_labels:
+            raise ValueError("x_tick_labels in sequence are inconsistent")
+
+    @staticmethod
+    def _from_spectra_bins_check(ref_bins_data: dict[str, Quantity],
+                                 spectrum: Spectrum2D) -> None:
+        """Check bin values and units match between spectra in new collection
+
+        Args:
+            ref_bins_data: ref axis names and values in format
+                ``{"x_data": Quantity(...), }``
+            spectrum: Item from sequence to compare with ref
+        """
+        for key, ref_bins in ref_bins_data.items():
+            item_bins = getattr(spectrum, key)
+            if not np.allclose(item_bins.magnitude, ref_bins.magnitude):
+                raise ValueError("Bins in sequence are inconsistent")
+            if item_bins.units != ref_bins.units:
+                raise ValueError("Bin units in sequence are inconsistent")
+
     @classmethod
-    def from_spectra(cls, spectra: Sequence[Spectrum2D]) -> Self:
+    def from_spectra(
+            cls, spectra: Sequence[Spectrum2D], unsafe: bool = False
```

Comment:
```suggestion
            cls, spectra: Sequence[Spectrum2D], *, unsafe: bool = False
```

Reply:
Argh, I saw you had done two and overlooked that one was the abstract method so there would be three 😅